exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -38,8 +44,12 @@
 // methods for compatibility.)
 class DrillStatementImpl extends AvaticaStatement implements DrillStatement,
                                                              DrillRemoteStatement {
+  //Not using the DrillbitContext's ExecutorService as this is threadPool is light-weight (threads wake up to cancel tasks) but needs a low response time
+  private static ExecutorService queryTimeoutTaskPool = Executors.newCachedThreadPool(new NamedThreadFactory("q-timeout-"));
```

Comment:
I'm not sure if there is a way for me to reference back to the Statement object... since the objective is to simply have a sleeping thread in this pool timeout and **cancel** the query. Let me see look around.